solana/integration-lib/src/lib.rs

```diff
@@ -51,7 +51,7 @@ impl Gateway {
     }
 
     /// Unpacks an account into a gateway token object
-    fn parse_gateway_token(account_info: &AccountInfo) -> Result<GatewayToken, GatewayError> {
+    pub fn parse_gateway_token(account_info: &AccountInfo) -> Result<GatewayToken, GatewayError> {
```

Comment:
I would push integrators to use the `InPlaceGatewayToken` api, it's far better on compute usage for no lost functionality. This could be converted to return that or just return `impl GatewayTokenAccess` to allow for future changes

Reply:
I suggest redesigning the API to export GatewayToken as an interface - then we can optimise the implementation without changing the API. But I won't add this to this PR as it would require a version change and a lot of refactoring probably.